pkg/controller/clone/planner.go

```diff
@@ -553,3 +738,63 @@ func createDesiredClaim(namespace string, targetClaim *corev1.PersistentVolumeCl
 
 	return desiredClaim
 }
+
+func createTempSourceClaim(ctx context.Context, namespace string, targetClaim *corev1.PersistentVolumeClaim, snapshot *snapshotv1.VolumeSnapshot, client client.Client) (*corev1.PersistentVolumeClaim, error) {
+	scName, err := getStorageClassNameForTempSourceClaim(ctx, snapshot, client)
+	if err != nil {
+		return nil, err
+	}
+	// Get the appropriate size from the snapshot
+	targetCpy := targetClaim.DeepCopy()
+	if snapshot.Status == nil || snapshot.Status.RestoreSize == nil || snapshot.Status.RestoreSize.Sign() == -1 {
+		return nil, fmt.Errorf("snapshot has no RestoreSize")
+	}
+	restoreSize := snapshot.Status.RestoreSize
+	if restoreSize.IsZero() {
+		reqSize := targetCpy.Spec.Resources.Requests[corev1.ResourceStorage]
+		restoreSize = &reqSize
+	}
+
+	desiredClaim := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   namespace,
+			Name:        fmt.Sprintf("tmp-source-pvc-%s", string(targetClaim.UID)),
+			Labels:      targetCpy.Labels,
+			Annotations: targetCpy.Annotations,
+		},
+		Spec: targetCpy.Spec,
+	}
+	desiredClaim.Spec.DataSource = nil
+	desiredClaim.Spec.DataSourceRef = nil
+	desiredClaim.Spec.StorageClassName = &scName
+	desiredClaim.Spec.Resources.Requests[corev1.ResourceStorage] = *restoreSize
+
+	return desiredClaim, nil
+}
+
+func getStorageClassNameForTempSourceClaim(ctx context.Context, snapshot *snapshotv1.VolumeSnapshot, client client.Client) (string, error) {
+	matches := []string{}
```

Comment:
nit: you don't need to initialize a zero length slice in order to work with it.  `var matches []string` would be the canonical definition in my experience.  But there is nothing wrong with this.